src/resources/uart_esp32.cc

```diff
@@ -178,6 +178,7 @@ class UartResource : public EventQueueResource {
   UART_ISR_INLINE void enable_tx_interrupts(bool begin);
   UART_ISR_INLINE void disable_tx_interrupts(bool done);
 
+  UART_ISR_INLINE bool has_flushed_tx_fifo();
```

Comment:
Perhaps `is_empty_tx_fifo()`? Somehow I think the "has flushed" could mean that it only returns true if you actively asked for flushing.

Reply:
done.